pkg/controller/hibernation/hibernation_controller.go

```
@@ -184,165 +184,160 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
-			return reconcile.Result{}, nil
-		case hivev1.StoppingHibernationReason, hivev1.HibernatingHibernationReason, hivev1.FailedToStartHibernationReason,
-			hivev1.InitializedConditionReason:
-			if controllerutils.IsFakeCluster(cd) {
+	if shouldHibernate || readyToHibernate {
+		// Signal a problem if we should be hibernating and the SyncSets have not yet been applied.
+		if clusterSyncExists && clusterSync.Status.FirstSuccessTime == nil {
+			// Allow hibernation (do not set condition) if hibernateAfterSyncSetsNotApplied duration has passed since cluster
+			// installed and syncsets still not applied
+			if cd.Status.InstalledTimestamp != nil && time.Now().Sub(cd.Status.InstalledTimestamp.Time) < hibernateAfterSyncSetsNotApplied {
+				return r.setHibernatingCondition(cd, hivev1.SyncSetsNotAppliedReason, "Cluster SyncSets have not been applied", corev1.ConditionFalse, cdLog)
+			}
+		}
+		// Hibernate fake cluster
+		if isFakeCluster && cd.Spec.HibernateAfter == nil && hibernatingCondition.Status != corev1.ConditionTrue {
+			r.setHibernatingCondition(cd, hivev1.HibernatingHibernationReason, "Fake cluster is stopped",
+				corev1.ConditionTrue, cdLog)
+		}
+		if readyToHibernate {
+			cd.Spec.PowerState = hivev1.HibernatingClusterPowerState
+			err := r.Update(context.TODO(), cd)
+			if err != nil {
+				cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error hibernating cluster")
+			}
+			return reconcile.Result{}, err
+		}
+		if shouldStopMachines(cd, hibernatingCondition) {
+			return r.stopMachines(cd, cdLog)
+		}
+		if hibernatingCondition.Reason == hivev1.StoppingHibernationReason {
+			return r.checkClusterStopped(cd, false, cdLog)
+		}
+	} else {
+		if shouldStartMachines(cd, hibernatingCondition) {
+			if isFakeCluster {
 				r.setHibernatingCondition(cd, hivev1.RunningHibernationReason, "Fake cluster is running",
 					corev1.ConditionFalse, cdLog)
-			} else {
-				return r.startMachines(cd, cdLog)
+				return reconcile.Result{}, nil
```

This should return the error if setting the status fails. Just return the result of `setHibernatingCondition`.